internal/kafka/internal/handlers/data_plane_cluster_test.go

```diff
@@ -0,0 +1,285 @@
+		fields         fields
+		wantStatusCode int
+	}{
+		{
+			name: "should return dataplane cluster config without an error",
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{
+					GetDataPlaneClusterConfigFunc: func(ctx context.Context, clusterID string) (*dbapi.DataPlaneClusterConfig, *errors.ServiceError) {
+						return &dbapi.DataPlaneClusterConfig{}, nil
+					},
+				},
+			},
+			wantStatusCode: http.StatusOK,
+		},
+		{
+			name: "should return an error if getting dataplane cluster config returns an error",
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{
+					GetDataPlaneClusterConfigFunc: func(ctx context.Context, clusterID string) (*dbapi.DataPlaneClusterConfig, *errors.ServiceError) {
+						return nil, errors.GeneralError("failed to get dataplane cluster config")
+					},
+				},
+			},
+			wantStatusCode: http.StatusInternalServerError,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewDataPlaneClusterHandler(tt.fields.dataplaneClusterService)
+			req, rw := GetHandlerParams("GET", "/{id}", nil)
+			req = mux.SetURLVars(req, map[string]string{"id": "test-id"})
+			h.GetDataPlaneClusterConfig(rw, req)
+			resp := rw.Result()
+			Expect(resp.StatusCode).To(Equal(tt.wantStatusCode))
+			resp.Body.Close()
+		})
+	}
+}
+
+func Test_validateBody(t *testing.T) {
+	type fields struct {
+		dataplaneClusterService services.DataPlaneClusterService
+	}
+
+	type args struct {
+		request *private.DataPlaneClusterUpdateStatusRequest
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		fields  fields
+		wantErr bool
+	}{
+		{
+			name: "should return no error with valid dataplane cluster status when validating body",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(nil), //&private.DataPlaneClusterUpdateStatusRequest{}, //SampleDataPlaneclusterStatusRequestWithAvailableCapacity(),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: false,
+		},
+		{
+			name: "should return an error with invalid Strimzi when validating body",
+			args: args{
+				request: dataplanemocks.BuildValidDataPlaneClusterUpdateStatusRequest(func(statusRequest *private.DataPlaneClusterUpdateStatusRequest) {
+					statusRequest.Strimzi = []private.DataPlaneClusterUpdateStatusRequestStrimzi{
+						{
+							Ready:            true,
+							Version:          "",
+							KafkaVersions:    []string{},
+							KafkaIbpVersions: []string{},
+						},
+					}
+				}),
+			},
+			fields: fields{
+				dataplaneClusterService: &services.DataPlaneClusterServiceMock{},
+			},
+			wantErr: true,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewDataPlaneClusterHandler(tt.fields.dataplaneClusterService)
```

Comment:
Let's add a `t.Parallel()` call here for parallel test execution

Reply:
done